modules/internal/src/main/scala/service.scala

```diff
@@ -207,16 +208,47 @@ object serviceImpl {
           )(implicit
           F: _root_.cats.effect.ConcurrentEffect[$F],
           EC: _root_.scala.concurrent.ExecutionContext
+        ): _root_.cats.effect.Resource[F, $Client[$F]] =
+          _root_.cats.effect.Resource.make {
+            val managedChannelInterpreter = new _root_.mu.rpc.client.ManagedChannelInterpreter[$F](channelFor, channelConfigList)
+            managedChannelInterpreter.build(channelFor, channelConfigList)
+          }(channel => F.delay(channel.shutdown())).flatMap(ch =>
+          _root_.cats.effect.Resource.make(F.delay(new $Client[$F](ch, options)))(_ => F.unit))
+        """.supressWarts("DefaultArguments")
+
+      val clientFromChannel: DefDef =
+        q"""
+        def clientFromChannel[$F_](
+          channel: F[_root_.io.grpc.ManagedChannel],
+          options: _root_.io.grpc.CallOptions = _root_.io.grpc.CallOptions.DEFAULT
+        )(implicit
+          F: _root_.cats.effect.ConcurrentEffect[$F],
+          EC: _root_.scala.concurrent.ExecutionContext
+        ): _root_.cats.effect.Resource[F, $Client[$F]] = _root_.cats.effect.Resource.make(channel)(channel =>
+        F.delay(channel.shutdown())).flatMap(ch =>
+        _root_.cats.effect.Resource.make(F.delay(new $Client[$F](ch, options)))(_ => F.unit))
+        """.supressWarts("DefaultArguments")
+
+      val unsafeClient: DefDef =
+        q"""
+        def unsafeClient[$F_](
+          channelFor: _root_.mu.rpc.ChannelFor,
+          channelConfigList: List[_root_.mu.rpc.client.ManagedChannelConfig] = List(
+            _root_.mu.rpc.client.UsePlaintext()),
+            options: _root_.io.grpc.CallOptions = _root_.io.grpc.CallOptions.DEFAULT
+          )(implicit
+          F: _root_.cats.effect.ConcurrentEffect[$F],
+          EC: _root_.scala.concurrent.ExecutionContext
         ): $Client[$F] = {
           val managedChannelInterpreter =
-            new _root_.mu.rpc.client.ManagedChannelInterpreter[F](channelFor, channelConfigList)
-          new $Client[$F](managedChannelInterpreter.build(channelFor, channelConfigList), options)
+            new _root_.mu.rpc.client.ManagedChannelInterpreter[$F](channelFor, channelConfigList)
+          new $Client[$F](managedChannelInterpreter.unsafeBuild(channelFor, channelConfigList), options)
         }""".supressWarts("DefaultArguments")
 
-      val clientFromChannel: DefDef =
+      val unsafeClientFromChannel: DefDef =
         q"""
-        def clientFromChannel[$F_](
-          channel: _root_.io.grpc.Channel,
+        def unsafeClientFromChannel[$F_](
+          channel: _root_.io.grpc.ManagedChannel,
```

Comment:
Since here we're not using the shutdown, I guess we can leave `channel: _root_.io.grpc.Channel,`